api/resources_portal/management/commands/material_additional_metadata_key_change.py

```diff
@@ -0,0 +1,119 @@
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from resources_portal.config.logging import get_and_configure_logger
+from resources_portal.models import Material
+
+logger = get_and_configure_logger(__name__)
+
+
+def validateCategory(category):
+    return len([choice for choice in Material.CATEGORY_CHOICES if category in choice]) > 0
+
+
+def addKey(category, new_key):
+    querySet = None
+    materials_changed = 0
+
+    if category:
+        if not validateCategory(category):
+            logger.error(f"The provided category '{category}' is not a valid material category.")
+            return
+        querySet = Material.objects.filter(category=category)
+    else:
+        querySet = Material.objects.all()
+
+    with transaction.atomic():
```

Comment:
I'm not sure we want to do this in a transaction. It'd be fine for now, but as we scale that could end up being a problem. Especially because this function is idempotent, so even if it is interrupted and has to be restarted it'd be fine right?

Reply:
David had mentioned doing this in a paginated fashion. I can go ahead and do so, though it'll be a little difficult

Reply:
Yeah you should just operate on slices of the query set at a time modify that slice then write that slice (page) to the db

Reply:
Just make sure that if you don't call `save()` on each object that you update `updated_at` manually because bulk updates won't do it.